lib/medtronic/medtronicSimulator.js

```diff
@@ -299,20 +349,25 @@ exports.make = function(config){
       setCurrBasal(event);
     },
     bolus: function(event) {
-      if(currWizard && currWizard.jsDate.valueOf() === event.jsDate.valueOf()) {
-        currWizard.bolus = event;
-        currWizard = currWizard.done();
-        simpleSimulate(currWizard);
+      if(currWizard) {
+        // A wizard event is always followed by a bolus event.
+        // Usually the timestamp is the same, but sometimes the bolus timestamp
+        // is a second later. Here we also check that the bolus event is at least
+        // within 5 seconds of the wizard event.
+        var bolusTime = event.jsDate.valueOf();
+        var wizardTime = currWizard.jsDate.valueOf();
+        if ( (bolusTime >= wizardTime) && ( bolusTime < (wizardTime + 5000)) ) {
```

Comment:
No, I didn't. The bolus record timestamp and wizard timestamp are usually exactly the same, but if they happen at the edge of a second the bolus timestamp may be recorded during the next second. I chose 5 seconds just to make sure that even if there's some short delay that somehow occurs between them they'll still match up. If you think this can cause potential issues, we could change it from 5s to 1s? (Note that this is not an instance of the wizard timestamp only being recorded after the initial bolus finishes, as the wizard record occurs first.)